.github/workflows/deploy-pr-preview.yaml

```diff
@@ -124,18 +128,34 @@ jobs:
           HEUREKA_CHANGED: ${{ contains(needs.run-detect-changes.outputs.changes, 'heureka') }}
           DOOP_CHANGED: ${{ contains(needs.run-detect-changes.outputs.changes, 'doop') }}
         run: |
```

Comment:
You can simplify this code by using an array to handle the components dynamically. This approach eliminates duplication and makes it much easier to add new apps in the future:

```yaml
run: |
  COMPONENTS=("ui-components" "supernova" "heureka" "doop")

  echo "<!DOCTYPE html><head><title>PR Preview</title></head><style> .failed { color: red; font-weight: bold; } </style><html><body><h1>Deployed Packages for PR ${GITHUB_HEAD_REF}</h1><ul>" > ${{ env.DEPLOY_PATH }}/index.html

  for component in "${COMPONENTS[@]}"; do
    CHANGED=$(echo "${{ contains(needs.run-detect-changes.outputs.changes, component) }}")
    BUILD_STEP="build-${component}"

    if [ "$CHANGED" = "true" ]; then
      if [ "${{ steps.${BUILD_STEP}.outcome }}" == "success" ]; then
        echo "<li><a href='./${component}/index.html'>${component^} (Success)</a></li>" >> ${{ env.DEPLOY_PATH }}/index.html
      else
        echo "<li class='failed'>${component^} (Failed)</li>" >> ${{ env.DEPLOY_PATH }}/index.html
      fi
    fi
  done

  echo "</ul></body></html>" >> ${{ env.DEPLOY_PATH }}/index.html
  ```

Reply:
oh! let me have a look. I didn't thought about it